src/main/clojure/clara/rules/engine.cljc

```diff
@@ -494,7 +494,9 @@
                                (str "with bindings\n  " bindings))
         message-header (string/join ["Condition exception raised.\n"
                                      "when processing fact\n"
-                                     (str "  " (pr-str fact) "\n")
+                                     (str "  " (->> (flatten [fact])
```

Comment:
As a general principle I think it almost always is a red flag to need to use `flatten`. What is arbitrary and out of control about the structure here that makes this necessary? 

If the problem is just that sometimes `fact` is not a fact, but rather a collection of facts, I think that should be handled explicitly to only  unroll that top level.

Better yet, I think the name should just be changed to `facts` and all callers should pass a collection. If they only have 1 they can put that in a coll too.

Reply:
Also is there any chance that this list of facts could be very large and cause a flood of output, eg. from accumulated coll values?

Reply:
@mrrodriguez I updated the PR title and description, I ended up rewriting the initial PR and targeting just the TestNode which is where there was missing exception handling logic; now, regarding the list of facts, this is the same possibility of flood of output with all the other Condition Exceptions where bindings are listed, because it is possible for the bindings used in a TestNode to be a list of accumulated facts all the same. I am wondering if I should change it so that the condition exception raised by the TestNode does not actually list facts because it does not really bind to any fact but it is purely dependent on the bindings and that is all that should be needed along with the condition/query encountering the exception.